src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/Overview/Cells/HealthCertifiedPerson/HealthCertifiedPersonTableViewCell.swift

```diff
@@ -411,7 +422,44 @@ class HealthCertifiedPersonTableViewCell: UITableViewCell, ReuseIdentifierProvid
 		bottomView.layer.borderColor = UIColor.enaColor(for: .cardBorder).cgColor
 		qrCodeContainerView.layer.borderColor = UIColor.enaColor(for: .cardBorder).cgColor
 	}
+	
+	private func fontColorForMaskState(maskStateIdentifier: MaskStateIdentifier) -> UIColor {
+		switch maskStateIdentifier {
+		case .maskRequired:
+			return .enaColor(for: .maskBadgeGrey)
+		case .maskOptional, .other:
+			return .enaColor(for: .textContrast)
+		}
+	}
+	
+	private func imageForMaskState(maskStateIdentifier: MaskStateIdentifier) -> UIImage? {
+		switch maskStateIdentifier {
+		case .maskRequired:
+			return UIImage(imageLiteralResourceName: "Icon_mask")
+		case .maskOptional:
+			return UIImage(imageLiteralResourceName: "Icon_noMask")
+		case .other:
+			return nil
+		}
+	}
 
+	private func gradientForMaskState(maskStateIdentifier: MaskStateIdentifier) -> GradientView.GradientType {
+		switch maskStateIdentifier {
+		case .maskRequired:
+			return .whiteWithGreyBorder
+		case .maskOptional, .other:
+			return .lightGreen
+		}
+	}
+	
+	private func gradientForAdmissionState(cellModel: HealthCertifiedPersonCellModel) -> GradientView.GradientType {
+		if cellModel.maskStateIdentifier == .maskOptional {
+			return .darkGreen
+		} else {
+			return cellModel.backgroundGradientType
+		}
+	}
+	
```

Comment:
I would move that stuff into the CellModel and get it from there. That matches more MVVM from my perspective.

Reply:
Good suggestion